src/plugins/dashboard/common/types.ts

```diff
@@ -43,62 +28,77 @@ export interface DashboardCapabilities {
 }
 
 /**
- * This should always represent the latest dashboard panel shape, after all possible migrations.
+ * The attributes of the dashboard saved object. This interface should be the
+ * source of truth for the latest dashboard attributes shape after all migrations.
  */
-export type SavedDashboardPanel = SavedDashboardPanel730ToLatest;
-
-export type SavedDashboardPanel640To720 = Pick<
-  RawSavedDashboardPanel640To720,
-  Exclude<keyof RawSavedDashboardPanel640To720, 'name'>
-> & {
-  readonly id: string;
-  readonly type: string;
-};
+export interface DashboardAttributes {
+  controlGroupInput?: RawControlGroupAttributes;
+  refreshInterval?: RefreshInterval;
+  timeRestore: boolean;
+  optionsJSON?: string;
+  useMargins?: boolean;
+  description: string;
+  panelsJSON: string;
+  timeFrom?: string;
+  version: number;
+  timeTo?: string;
+  title: string;
+  kibanaSavedObjectMeta: {
+    searchSourceJSON: string;
+  };
+}
 
-export type SavedDashboardPanel630 = Pick<
-  RawSavedDashboardPanel630,
-  Exclude<keyof RawSavedDashboardPanel620, 'name'>
-> & {
-  readonly id: string;
-  readonly type: string;
-};
+/** --------------------------------------------------------------------
+ * Dashboard panel types
+ -----------------------------------------------------------------------*/
 
-export type SavedDashboardPanel620 = Pick<
-  RawSavedDashboardPanel620,
-  Exclude<keyof RawSavedDashboardPanel620, 'name'>
-> & {
-  readonly id: string;
-  readonly type: string;
-};
+/**
+ * The dashboard panel format expected by the embeddable container.
+ */
+export interface DashboardPanelState<
+  TEmbeddableInput extends EmbeddableInput | SavedObjectEmbeddableInput = SavedObjectEmbeddableInput
+> extends PanelState<TEmbeddableInput> {
+  readonly gridData: GridData;
+  panelRefName?: string;
+}
 
-export type SavedDashboardPanel610 = Pick<
-  RawSavedDashboardPanel610,
-  Exclude<keyof RawSavedDashboardPanel610, 'name'>
-> & {
-  readonly id: string;
-  readonly type: string;
+/**
+ * A saved dashboard panel parsed directly from the Dashboard Attributes panels JSON
+ */
+export type SavedDashboardPanel = SerializableRecord & {
```

Comment:
While on the surface having a type that guarantees that data is serialisable is a good thing, we should be very careful with the `SerializableRecord` or `Serializable` types. These are really anti-types in that they reduce type safety making code more prone to bugs (especially bad in sensitive code like this).

As an example `mySavedDashboardPanel.recordDoesNotExist.fieldDoesNotExist = 'some string'` would happily pass the type checker. So it really has the same problems as using `any`.

I know this PR didn't introduce this type to the dashboard code, but if we could limit/reduce it's usage that improve the quality of the code.

Reply:
Hey @rudolf, this is a really good point! Let me look into removing any instances of `SerializableRecord` from Dashboard. I wonder if there are any Typescript extension types that we can use to enforce serializability without introducing that nasty key signature.

Reply:
I've removed a couple of references here, but removing all of the `SavedObjectAttributes` imports from Dashboard is proving to be a big enough project to warrant its own follow-up PR. I have created [this issue](https://github.com/elastic/kibana/issues/140983) to track it